packages/jest-types/src/Global.ts

```diff
@@ -1,101 +1,19 @@
-/**
- * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
- *
- * This source code is licensed under the MIT license found in the
- * LICENSE file in the root directory of this source tree.
- */
-
-import {CoverageMapData} from 'istanbul-lib-coverage';
-
-export type DoneFn = (reason?: string | Error) => void;
-export type TestName = string;
-export type TestFn = (done?: DoneFn) => Promise<any> | void | undefined;
-export type BlockFn = () => void;
-export type BlockName = string;
-
-export type Col = unknown;
-export type Row = Array<Col>;
-export type Table = Array<Row>;
-export type ArrayTable = Table | Row;
-export type TemplateTable = TemplateStringsArray;
-export type TemplateData = Array<unknown>;
-export type EachTable = ArrayTable | TemplateTable;
-export type EachTestFn = (
-  ...args: Array<any>
-) => Promise<any> | void | undefined;
-
-// TODO: Get rid of this at some point
-type Jasmine = {_DEFAULT_TIMEOUT_INTERVAL?: number; addMatchers: Function};
-
-type Each = (
-  table: EachTable,
-  ...taggedTemplateData: Array<unknown>
-) => (title: string, test: EachTestFn, timeout?: number) => void;
-
-export interface ItBase {
-  (testName: TestName, fn: TestFn, timeout?: number): void;
-  each: Each;
-}
-
-export interface It extends ItBase {
-  only: ItBase;
-  skip: ItBase;
-  todo: (testName: TestName, ...rest: Array<any>) => void;
-}
-
-export interface ItConcurrentBase {
-  (testName: string, testFn: () => Promise<any>, timeout?: number): void;
-}
-
-export interface ItConcurrentExtended extends ItConcurrentBase {
-  only: ItConcurrentBase;
-  skip: ItConcurrentBase;
-}
-
-export interface ItConcurrent extends It {
-  concurrent: ItConcurrentExtended;
-}
-
-export interface DescribeBase {
-  (blockName: BlockName, blockFn: BlockFn): void;
-  each: Each;
-}
-
-export interface Describe extends DescribeBase {
-  only: DescribeBase;
-  skip: DescribeBase;
-}
-
-// TODO: Maybe add `| Window` in the future?
-export interface Global extends NodeJS.Global {
-  it: ItConcurrent;
-  test: ItConcurrent;
-  fit: ItBase & {concurrent?: ItConcurrentBase};
-  xit: ItBase;
-  xtest: ItBase;
-  describe: Describe;
-  xdescribe: DescribeBase;
-  fdescribe: DescribeBase;
-  __coverage__: CoverageMapData;
-  jasmine: Jasmine;
-  fail: () => void;
-  pending: () => void;
-  spyOn: () => void;
-  spyOnProperty: () => void;
-}
+import { It, ItConcurrent, ItBase, Describe, DescribeBase } from './Global-sans-augments';
+import { Jasmine } from './Jasmine';
+export * from './Global-sans-augments';
 
 declare global {
```

Comment:
we should not be messing with the global at all, could you just remove this entire `declare global {` block? And rollback the other changes in this PR. We might want/need some kind of `JestGlobalEnv`:

```ts
interface JestGlobal extends NodeJS.Global {
  // ...blahblahblah
}
```

but it should not mutate the actual global type environment

---

The TS types are mostly internal for now, they are not meant to replace `@types/jest` yet. When  they are it'll be the `jest` package itself sticking things on `global`

Reply:
I haven't used jest in a little while, but `describe()` is available globally, right?  As in, users do not need to `import`.  If so it needs to be declared globally *somewhere* for users of Jest.  But I don't know if you want that to be happening within `@jest/types` or somewhere else.

Reply:
I missed your last sentence about omitting the global declarations for now and eventually moving them into the jest package.  Roger that.

Is there any harm in me moving the globals to the jest package right away?  If people are still using @types/jest, it'll be ignored.  But if they skip that and use the types bundled in jest, it'll pick up the global declaration and everything will work.

Or should it *always* be opt-in to get the global declarations?  E.g. people can `import 'jest';` without the global declarations, and they'll need to explicitly `/// <reference types="jest/globals" />` to get the global types?  Will anyone ever want to `import jest from 'jest';` to use the API, without getting the global declarations?

Reply:
For now, the way to get the global type defs with Jest should be `@types/jest`. THat will change at some point, but _for now_ we do not support patching the global types

Reply:
@SimenB Have you hit issues where the Typescript compiler uses `node_modules/jest` types, skipping `node_modules/@types/jest`?  That's the behavior I'm seeing.  It doesn't matter if the user installs a `@types` package.  It'll be ignored if the non-`@types` package bundles its own declarations.
https://repl.it/@AndrewBradley/SpanishHugeVirus-1

Reply:
no, but you can use 

```json
{
  "compilerOptions": {
    "paths": {
      "jest": ["./node_modules/@types/jest", "./node_modules/jest"]
    }
  }
}
```

or something like that to prioritize DT's typings